### ISSUE TITLE
wikicreole parser

### ISSUE THREAD

**Issue by jgm** [jgm/pandoc#195]
<pre>
There's a WikiCreole project out there which has the potential to unify the
syntax of the diverse wiki engines out there (and new wikis to come)

See http://www.wikicreole.org/

What's interesting about this project is that they're not just wading in
and proposing some whole new markup language.  OK, they *are* proposing a
new markup language, but this time, they are being careful to research
every decision based on existing wikis, reason about each choice an
document said reasoning.  

This does not appear to be some fly-by-night operation either; it has
participation from  some MediaWiki folks (Brion Vibber) and other wiki
authors, notably Ward Cunningham.

Note that the project has produced a 1.0 version of their grammar which has
been frozen since 2007-07-04, so writing this parser would just be work and
not head-scratching (as it is with the mediawiki syntax).

One interesting new bit of syntax is the notion of 'placeholders' which
allow untranslatable bits of legacy markup to be stored in Creole to make
it feasible to use as much Creole as possible within a wiki that has
fancier markup.

I don't personally have a use for Creole right now, but I imagine the
Pandoc team would be interested to hear about it, and that the Creole folks
will be happy to know about this cool migration tool Pandoc that can help
us achieve convergence one day :-)

Google Code Info:
Issue #: 160
Author: eric....@gmail.com
Created On: 2009-08-10T15:14:59.000Z
Closed On: 
</pre>

**Comment by grexe** [comment]
I could live with that, frankly, but I'd really prefer to have the cleaner Creole syntax than the native syntax of DokuWiki in my case (first thing I did was to install the Creole plugin;).
Don't want to start a discussion, but the formatting for headings and other formatting is just so straight-forward (resembling the good old ASCII formatting rules used in pre-HTML mail...)...

Any chance Creole support makes it into pandoc?
Great tool btw, thanks very much!

**Comment by layus** [comment]
@jgm I just wanted to point out https://github.com/tdammers/pandoc-creole. Any chance to get that merged in master ?
Also, see duplicate/related #2292 and #524.

I have no time to implement this, but would be really happy to have Creole support in our beloved pandoc :-).

**Comment by jgm** [comment]
Question is how complete the pandoc-creole parser is.
If it's pretty complete, then perhaps you should ask the
author whether they have any plans to submit to pandoc?

**Comment by layus** [comment]
@jgm Right, and done. IFAIK, its only a creole reader/parser, but its a good start.

**Comment by tdammers** [comment]
pandoc-creole author chiming in here.

I wrote that library for a very specific personal use case, namely adding creole support in order to migrate an existing website (running on a custom PHP backend) to a pandoc-based platform (https://github.com/tdammers/sprinkles, to be precise). The PHP platform in question has some nonstandard extensions to Creole, which I added to the parser (specifically, it allows annotating various syntax elements with CSS classes using the nonstandard `@(classname)` syntax); the parsers that handle this start about [here](https://github.com/tdammers/pandoc-creole/blob/master/src/Text/Pandoc/Readers/Creole.hs#L70).

Also, I didn't implement all of Creole, only the parts I needed right there and then, because I am not interested in adding more Creole content - anything new will use Markdown, because in terms of general adoption, it seems to be the winner, at least for the foreseeable future.

With all this in mind, I am not at all opposed against merging the Creole parser into Pandoc core, but I am pretty certain it would need some more work before it's ready for prime time. If anyone feels like picking up this particular ball, I'd be happy to help, but I'm afraid I can't spare the time to do it alone. Then again, having it as an add-on library like it is now isn't too bad a situation either, and it would keep pandoc core smaller. I'd be totally OK with either situation.
